network/datadog_checks/network/network.py

```diff
@@ -248,7 +249,12 @@ def _submit_devicemetrics(self, iface, vals_by_metric, tags):
         expected_metrics = self._get_expected_metrics()
         for m in expected_metrics:
             assert m in vals_by_metric
-        assert len(vals_by_metric) == len(expected_metrics)
+        if len(vals_by_metric) > len(expected_metrics):
+            unexpected_metrics = set(vals_by_metric.keys()).difference(expected_metrics)
+            raise CheckException("Unexpected metrics found: {}".format(unexpected_metrics))
+        if len(vals_by_metric) < len(expected_metrics):
+            missing_metrics = set(expected_metrics).difference(vals_by_metric.keys())
+            raise CheckException("Missing expected metrics: {}".format(missing_metrics))
```

Comment:
I'd still probably prefer to have the assertion rather than picking some other exception for this, maybe something like this:
```suggestion
        error_msg = ""
        if len(vals_by_metric) > len(expected_metrics):
            unexpected_metrics = set(vals_by_metric.keys()).difference(expected_metrics)
            error_msg = "Unexpected metrics found: {}".format(unexpected_metrics))
        elif len(vals_by_metric) < len(expected_metrics):
            missing_metrics = set(expected_metrics).difference(vals_by_metric.keys())
            error_msg = CheckException("Missing expected metrics: {}".format(missing_metrics))
            
        assert len(vals_by_metric) == len(expected_metrics), error_msg
```

Reply:
CheckException and ConfigurationError are the standard exceptions checks should be raising

Reply:
My bad, I thought I was looking at a test due to the original assertion.